File: dart/lib/profiles.dart
import 'dart:convert';
import 'package:flutter/services.dart' show rootBundle;
import 'types.dart';

Future<Map<String, List<Link>>> loadJsonAssets() async {
  final Map<String, List<Link>> models = {};
  final List<String> jsonFiles = [
    ...json
        .decode(await rootBundle.loadString('AssetManifest.json'))
        .keys
        .where((String key) => key.startsWith('../patterns/'))
        .map((String path) => path.split('/').last)
        .toList(),
  ];

  for (final file in jsonFiles) {
    final content = await rootBundle.loadString('../patterns/$file');
    final Map<String, dynamic> jsonData = json.decode(content);
    models[Profile.fromJson(jsonData).name] =
        Profile.fromJson(jsonData).matches;
  }
  return models;
}

File: ts/src/profiles.ts
import { Profile, Link } from './types.ts';
import { patterns } from './generated/patterns.ts';

export function loadProfiles(): Record<string, Link[]> {
  const models: Record<string, Link[]> = {};

  Object.values(patterns).forEach((profile) => {
    const _profile = profile as Profile;
    models[_profile.name] = _profile.matches.map(match => ({
      ...match,
      idPattern: match.idPattern || _profile.pattern,
      forceStripQuery: match.forceStripQuery || _profile.forceStripQuery,
      icon: _profile.icon
    }));
  });

  return models;
} 